packages/grid/x-grid/src/tests/columns.XGrid.test.tsx

```diff
@@ -75,4 +87,87 @@ describe('<XGrid /> - Columns', () => {
       await waitFor(() => expect(screen.queryByRole('menu')).to.equal(null));
     });
   });
+
+  describe('resizing', () => {
+    const columns = [{ field: 'brand', width: 100 }];
+
+    let clock;
+
+    beforeEach(() => {
+      clock = useFakeTimers();
+    });
+
+    afterEach(() => {
+      clock.restore();
+    });
+
+    it('should allow to resize columns with the mouse', () => {
+      render(<Test columns={columns} />);
+      const separator = document.querySelector(
+        `.${GRID_COLUMN_HEADER_SEPARATOR_RESIZABLE_CSS_CLASS}`,
+      );
+      fireEvent.mouseDown(separator, { clientX: 100 });
+      fireEvent.mouseMove(separator, { clientX: 110, buttons: 1 });
+      fireEvent.mouseUp(separator);
+      expect(getColumnHeaderCell(0).style.width).to.equal('110px');
+      expect(getCell(1, 0).style.width).to.equal('110px');
+    });
+
+    it('should allow to resize columns with the touch', () => {
+      render(<Test columns={columns} />);
+      const separator = document.querySelector(
+        `.${GRID_COLUMN_HEADER_SEPARATOR_RESIZABLE_CSS_CLASS}`,
+      )!;
+      const now = Date.now();
+      const touchStartEvent = createEvent.touchStart(separator);
+      // Safari lacks support for constructing a TouchEvent with changedTouches
+      Object.defineProperty(touchStartEvent, 'changedTouches', {
```

Comment:
Could we use the native browser API instead? See the tests in the slider: https://github.com/mui-org/material-ui/blob/0230df94ca877fdd63a42c3e276f75c3a4316eea/packages/material-ui/src/Slider/Slider.test.js#L24

forcing a property looks suspicious
```suggestion
```